src/main/java/mekanism/common/tile/prefab/TileEntityMultiblock.java

```diff
@@ -113,12 +113,13 @@ protected boolean onUpdateServer() {
             structure.tick(this, ticker % MekanismUtils.TICKS_PER_HALF_SECOND == 0);
         }
         T multiblock = getMultiblock();
-        if (isMaster() && multiblock.isFormed() && multiblock.recheckStructure) {
-            multiblock.recheckStructure = false;
-            getStructure().doImmediateUpdate(this, ticker % MekanismUtils.TICKS_PER_HALF_SECOND == 0);
-            multiblock = getMultiblock();
-        }
         if (multiblock.isFormed()) {
+            if (isMaster() && multiblock.recheckStructure) {
+                multiblock.recheckStructure = false;
+                getStructure().doImmediateUpdate(this, ticker % MekanismUtils.TICKS_PER_HALF_SECOND == 0);
+                multiblock = getMultiblock();
```

Comment:
Does moving this inside the formed block cause problems? (Logic is a change for if the ending multiblock is not formed, this is a different code path than it was before.